### PR TITLE
[LC-1000] Distinguish Consensus essential data and Create backup db.

### PR DESCRIPTION
**Pull request by windies21**
Distinguish Consensus essential data.
The node must be able to request and deliver consensus essential data.
Organize additional items for background sync and fast start
Separate essential data and check normal operation of existing node using separated data
Requesting essential data with cli to create backup db (loopchain_tools)

### CODE REVIEW

File: tools/basictool.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import os
+import traceback
+from collections import OrderedDict
+from pathlib import Path
+from sys import platform
+
+from earlgrey import MessageQueueService
+from tools import get_option_from_prompt
+
+from testcase.unittest.test_util import clean_up_temp_db_files, clean_up_mq
```
Comment (metalg0su) [inline tools/basictool.py]: Looks weird to use test helper located at test dir.

How about re-locate them?

File: tools/basictool.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+"""CLI tool for loopchain"""
+import inspect
+import os
+import traceback
+from collections import OrderedDict
+from pathlib import Path
+from sys import platform
+
+from earlgrey import MessageQueueService
```
Comment (metalg0su) [inline tools/basictool.py:9]: `Earlgrey` is not in requirements.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    'license': "Apache License 2.0",
+    'install_requires': list(open('requirements.txt')),
+    'classifiers': [
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'Intended Audience :: System Administrators',
+        'Natural Language :: English',
+        'License :: OSI Approved :: Apache Software License',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 3.6',
```
Comment (metalg0su) [inline setup.py]: What is supported python versions?

Loopchain limits the python version above 3.7.
And as its tool `loopchain-tools` has different version range.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+setup_options = {
+    'name': 'loopchain tools',
+    'description': 'CLI tools for loopchain',
+    'long_description': open('README.md').read(),
+    'long_description_content_type': 'text/markdown',
+    'url': 'https://github.com/icon-project/loopchain_tools',
+    'version': version,
+    'author': 'ICON foundation',
+    'author_email': 'foo@icon.foundation',
```
Comment (metalg0su) [inline setup.py]: Is it valid address?

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    'packages': find_packages(),
+    'license': "Apache License 2.0",
+    'install_requires': list(open('requirements.txt')),
+    'classifiers': [
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'Intended Audience :: System Administrators',
+        'Natural Language :: English',
+        'License :: OSI Approved :: Apache Software License',
```
Comment (metalg0su) [inline setup.py:28]: This project has no LICENSE documentation.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Loopchain Tools
+ **Loopchain tools** is a Loopchain tool.
+
+The tool is consisted of two parts:
+- [Basictool]: provides configure preset files and supports variable API methods for Loopchain
+- [Gtool]: provides CLI monitoring services.
+
+## Getting Started
+### Installation
+ Loopchain Tools has no dependencies on Loopchain itself.
```
Comment (metalg0su) [inline README.md]: How can I run Loopchain Tools itself, if it does not require Loopchain?

Can you explain?

File: tools/basictool.py (hunk 1 above)
Comment (windies21) [reply tools/basictool.py]: 'basictool.py' is the same as the private version(=demotool.py). So I have no interest about refactoring now.

Comment (windies21) [reply tools/basictool.py]: The goal of this PR is to add a new single feature. And to make it open source. After then we will discuss about refactoring.

File: tools/basictool.py (hunk 2 above)
Comment (windies21) [reply tools/basictool.py:9]: This tool cannot be run independently. This tool run in loopchain's venv with some additional requirements. so Its requirements has only additionals.

File: README.md (hunk 6 above)
Comment (windies21) [reply README.md]: This branch is just copy of private One. So It needs another work to making it public. If you want it in this PR?

Comment (windies21) [reply README.md]: I just made copy one and add a  new single feature. I didn't touch anything.

Comment (windies21) [reply README.md]: You want it? retouch everything?

Comment (windies21) [reply README.md]: That was not  my Goal.

Comment (windies21) [reply README.md]: But I will reconsider it.

**Review by metalg0su — APPROVED** [review]
LGTMe

**Pull request merged**